src/test/java/mate/academy/dao/impl/UserDaoImplTest.java

```diff
@@ -0,0 +1,83 @@
+package mate.academy.dao.impl;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import java.util.Optional;
+import java.util.Set;
+import mate.academy.dao.AbstractTest;
+import mate.academy.dao.RoleDao;
+import mate.academy.dao.UserDao;
+import mate.academy.model.Role;
+import mate.academy.model.User;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+class UserDaoImplTest extends AbstractTest {
+    private static final String EMAIL = "bob@i.ua";
+    private static final String INVALID_EMAIL = "alice@i.ua";
+    private static final String PASSWORD = "1234";
+    private static final Long ID = 1L;
+    private static final Long INVALID_ID = 2L;
+    private UserDao userDao;
+    private RoleDao roleDao;
+    private User user;
+    private Role role;
+
+    @Override
+    protected Class<?>[] entities() {
+        return new Class[]{User.class, Role.class};
+    }
+
+    @BeforeEach
+    void setUp() {
+        userDao = new UserDaoImpl(getSessionFactory());
+        roleDao = new RoleDaoImpl(getSessionFactory());

```

Comment:
It feels strange to create a new object before every test. It can be initialized in @BeforeAll or just on 24-25 lines